src/transformers/tokenization_utils_base.py

```diff
@@ -2869,6 +2872,12 @@ def pad(
             verbose (`bool`, *optional*, defaults to `True`):
                 Whether or not to print more information and warnings.
         """
+        if self.__class__.__name__.endswith("Fast"):
+            warnings.warn(
```

Comment:
I use `warnings.warn` instead of `logger.warning` so that the warning can be issued only once.